Client/src/Components/DatasetUpload/XLSdatasetDownloadButton.tsx

```diff
@@ -0,0 +1,136 @@
+import { Button } from '@material-ui/core'
+import { IDatasetModel } from '../../Models/Datasets/IDatasetModel'
+import { useDimensionsSelector } from '../../Stores/Slices/DimensionsSlice'
+import { getUnitNameById } from '../../Common/Helpers/DimensionHelpers'
+import * as XLSX from 'xlsx';
+import { listCategories } from '../../Remote/Endpoints/CategoryEndpoint'
+import React, { useEffect, useState } from 'react'
+
+interface IProps {
+    datasets: IDatasetModel
+}
+
+export const XLSdatasetDownloadButton = (props: IProps) => {
+    const { datasets } = { ...props }
+    const dimensions = useDimensionsSelector()
+    const [categories, setCategories] = useState([])
+
+    useEffect(() => {
+        const callListCategories = async () => {
+            const getCategories = await listCategories()
+            setCategories(getCategories || [])
+        }
+        callListCategories()
+    }, [])
+
+
+    const materialArray = (): any => {
+        var name = []
+        name.push("Material")
+        for (var i in datasets.material) {
+            name.push(datasets.material[i].composition + " \'" + datasets.material[i].details + "\'")
+        }
+        return name
+    }
+    const authorArray = (): any => {
+        var name = []
+        name.push("Author")
+        for (var i in datasets.reference.authors) {
+            if (datasets.reference.authors[i].middleName) {
+                name.push(datasets.reference.authors[i].firstName + " " + datasets.reference.authors[i].middleName + " " + datasets.reference.authors[i].lastName)
+            }
+            else {
+                name.push(datasets.reference.authors[i].firstName + " " + datasets.reference.authors[i].lastName)
+
+            }
+        }
+        return name
+    }
+    const variableArray = (): any => {
+        var name = []
+        for (var i in datasets.data.variables) {
+            name.push(datasets.data.variables[i].name + "  [" + getUnitNameById(dimensions, datasets.data.variables[i].unitId) + "] ")
+        }
+        name.push("comments")
+        return name
+    }
+    const categoryName = () => {
+        let cgName = []
+        cgName.push("Categories")
+        categories.map(category => {
+            if (category.id == datasets.category) {
+                cgName.push(category.name)
+            }
+        })
+        return cgName
+    }
+    const subcategoryName = () => {
+        let scgName = []
+        let cat
+        scgName.push("Subcategories")
+        categories.map(category => {
+            if (category.id == datasets.category) {
+                cat = category
+                cat.subcategories.map(subcategory => {
+                    if (subcategory.id == datasets.subcategory) {
+                        scgName.push(subcategory.name)
+                    }
+                })
+            }
+        })
+        return scgName
+    }
+    const compileDataArray = (): any => {
```

Comment:
Slightly messy, but it works. Without time constraints something cleaner could be looked into, but alas